src/components/drawer/drawer.test.tsx

```diff
@@ -0,0 +1,480 @@
+import React from "react";
+import { act, render, screen, waitFor } from "@testing-library/react";
+
+import userEvent from "@testing-library/user-event";
+import Drawer from ".";
+
+describe("Drawer", () => {
+  beforeEach(() => {
+    jest.useFakeTimers();
+  });
+
+  afterEach(() => {
+    jest.runOnlyPendingTimers();
+    jest.useRealTimers();
+  });
+
+  it("cleans ups timers on unmount", async () => {
+    const clearTimeoutSpy = jest.spyOn(window, "clearTimeout");
+    const { unmount } = render(<Drawer>Foobar</Drawer>);
+
+    unmount();
+
+    await waitFor(() => {
+      expect(screen.queryByTestId("drawer")).not.toBeInTheDocument();
+    });
+    expect(clearTimeoutSpy).toHaveBeenCalledTimes(1);
+
+    clearTimeoutSpy.mockRestore();
+  });
+
+  it("sets correct data-* props on main dialog element", () => {
+    render(
+      <Drawer data-element="foo" data-role="bar">
+        Foobar
+      </Drawer>
+    );
+    const drawer = screen.getByTestId("bar");
+
+    expect(drawer).toHaveAttribute("data-component", "drawer");
+    expect(drawer).toHaveAttribute("data-element", "foo");
+  });
+
+  it("has accessible name when ariaLabel prop is provided", () => {
+    render(<Drawer aria-label="test">Foobar</Drawer>);
+    expect(screen.getByTestId("drawer")).toHaveAccessibleName("test");
+  });
+
+  it("renders with 100% height by default", () => {
+    render(<Drawer>Foobar</Drawer>);
+    expect(screen.getByTestId("drawer")).toHaveStyle({ height: "100%" });
+  });
+
+  it("renders with custom height when height prop is provided", () => {
+    render(<Drawer height="50%">Foobar</Drawer>);
+    expect(screen.getByTestId("drawer")).toHaveStyle({ height: "50%" });
+  });
+
+  it("displays heading when title prop is provided", () => {
```

Comment:
question (non-blocking): are all these style tests needed for coverage? If not we could probably rely on chromatic and playwright as we can control the open/closed state etc

Reply:
I'll try removing these style tests and see if the coverage is affected :+1: